### PR TITLE
Feat/branch get compare command

### PR DESCRIPTION
**Pull request by almeida-shrutika**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=0-c1hzZ)

Changes:
- Implement branch config command
- Implement branch list command

### CODE REVIEW

File: packages/contentstack-branch/src/commands/cm/branches/config.ts [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
-  static examples: string[] = ['csdx cm:branches:config --base-branch main --stack-api-key bltxxxxxxxx']; // Note: add more examples
+  static examples: string[] = [
+    'csdx cm:branches:config',
+    'csdx cm:branches:config -k <stack api key> --base-branch <base branch>',
+    'csdx cm:branches:config --global -k <stack api key> --base-branch <base branch>',
+    'csdx cm:branches:config --base-branch <base branch>',
+  ]; // Note: add more examples
 
-  static usage: string = 'cm:branches:config [--base-branch <value>] [--stack-api-key <value>]'; // Note: add all flags
+  static usage: string = 'cm:branches:config [-a <value>] [--base-branch <value>] [--stack-api-key <value>] [-y]'; // Note: add all flags
```
Comment (shafeeqd959) [inline packages/contentstack-branch/src/commands/cm/branches/config.ts:15]: pls remove flags that are not part of config plugin

File: packages/contentstack-branch/src/commands/cm/branches/config.ts [hunk 2]
```diff
@@ -1,24 +1,63 @@
...
+        configHandler.set(`base-branch`, { baseBranch, apiKey });
+      } else {
+        let branchJson = JSON.stringify({
+          'base-branch': {
+            apiKey: apiKey,
+            baseBranch: baseBranch,
+          },
+        });
+
+        writeFile(__dirname + '/branch-config.json', branchJson, (err) => {
```
Comment (shafeeqd959) [inline packages/contentstack-branch/src/commands/cm/branches/config.ts:50]: use a write file sync method instead

File: packages/contentstack-branch/src/commands/cm/branches/config.ts [hunk 3]
```diff
@@ -1,24 +1,63 @@
...
+          if (err) {
+            console.error('Failed to write config file: ', err);
+          } else console.log('Config file: branch-config.json');
```
Comment (shafeeqd959) [inline packages/contentstack-branch/src/commands/cm/branches/config.ts:53]: Give full path to the location

File: packages/contentstack-branch/src/commands/cm/branches/config.ts [hunk 4]
```diff
@@ -1,24 +1,63 @@
...
+        });
+      }
+
+      cliux.print('Base Branch: ' + baseBranch, { color: 'blue' });
+      cliux.print('Stack Api Key: ' + apiKey, { color: 'blue' });
```
Comment (shafeeqd959) [inline packages/contentstack-branch/src/commands/cm/branches/config.ts:58]: Print a success message

File: packages/contentstack-branch/src/commands/cm/branches/index.ts [hunk 5]
```diff
@@ -1,24 +1,81 @@
...
+
+      if (existsSync(__dirname + '/branch-config.json')) {
+        let data = JSON.parse(readFileSync(__dirname + '/branch-config.json', 'utf-8'));
+        config = data['base-branch'];
+      } else if (Boolean(configHandler.get(`base-branch`))) {
+        config = configHandler.get(`base-branch`);
+      }
+
+      if (config && Object.keys(config).length > 0) {
+        const branchResult = await managementAPIClient.stack({ api_key: config.apiKey }).branch().query().find();
```
Comment (shafeeqd959) [inline packages/contentstack-branch/src/commands/cm/branches/index.ts:35]: if the config not set this code will print branch is not available, which is wrong. Expectation is it needs to fetch the branch list always and to identify the current base branch logic needs to make use of the config.

**Comment by shafeeqd959** [comment]
@almeida-shrutika , update your branch with latest, base branch has changed

**Comment by netrajpatel** [comment]
@almeida-shrutika Some of the snyk test workflows are failing for this PR, please fix those.

**Comment by almeida-shrutika** [comment]
> @almeida-shrutika Some of the snyk test workflows are failing for this PR, please fix those.

@netrajpatel this PR is not needed anymore so will close this PR.

**Pull request closed without merging**